extensions/amp-social-share/0.2/amp-social-share.js

```diff
@@ -0,0 +1,178 @@
+    // Hide/ignore component if typeConfig is undefined
+    if (!typeConfig) {
+      toggle(this.element, false);
+      return;
+    }
+    this.setHrefAndTargetContext_(typeConfig, platform);
+  }
+
+  /** @override */
+  isLayoutSupported() {
+    userAssert(
+      isExperimentOn(this.win, 'amp-social-share-v2'),
+      'expected amp-social-share-v2 experiment to be enabled'
+    );
+    return true;
+  }
+
+  /**
+   * @private
+   * @param {?string} type
+   * @param {!../../../src/service/viewer-interface.ViewerInterface} viewer
+   * @param {!../../../src/service/platform-impl.Platform} platform
+   * @return {!JsonObject|undefined}
+   */
+  getTypeConfigOrUndefined_(type, viewer, platform) {
+    userAssert(type, 'The type attribute is required. %s', this.element);
+    userAssert(
+      !/\s/.test(type),
+      'Space characters are not allowed in type attribute value. %s',
+      this.element
+    );
+    if (type === 'system') {
+      // navigator.share unavailable
+      if (!this.systemShareSupported_(viewer, platform)) {
+        return;
+      }
+    } else {
+      // system share wants to be unique
+      const systemOnly =
+        this.systemShareSupported_(viewer, platform) &&
+        !!window.document.querySelector(
+          'amp-social-share[type=system][data-mode=replace]'
+        );
+      if (systemOnly) {
+        return;
+      }
+    }
+    const typeConfig = getSocialConfig(type) || dict();
+    if (typeConfig['obsolete']) {
+      user().warn(TAG, `Skipping obsolete share button ${type}`);
+      return;
+    }
+    return typeConfig;
+  }
+
+  /**
+   * @private
+   * @param {!../../../src/service/viewer-interface.ViewerInterface} viewer
+   * @param {!../../../src/service/platform-impl.Platform} platform
+   * @return {boolean}
+   */
+  systemShareSupported_(viewer, platform) {
+    // Chrome exports navigator.share in WebView but does not implement it.
+    // See https://bugs.chromium.org/p/chromium/issues/detail?id=765923
+    const isChromeWebview = viewer.isWebviewEmbedded() && platform.isChrome();
+
+    return 'share' in navigator && !isChromeWebview;
+  }
+
+  /**
+   * Sets 'href' and 'target' in the AmpContext based on AMP-specific conditions.
+   * These needs to be passed to the context as opposed to as properties in `init()`
+   * because their values are resolved asynchronously.
+   *
+   * @private
+   * @param {!JsonObject} typeConfig
+   * @param {!../../../src/service/platform-impl.Platform} platform
+   */
+  setHrefAndTargetContext_(typeConfig, platform) {
+    const shareEndpoint = userAssert(
+      this.element.getAttribute('data-share-endpoint') ||
+        typeConfig['shareEndpoint'],
+      'The data-share-endpoint attribute is required. %s',
+      this.element
+    );
+    const urlParams = getDataParamsFromAttributes(this.element);
+    Object.assign(urlParams, typeConfig['defaultParams']);
+    const hrefWithVars = addParamsToUrl(
+      dev().assertString(shareEndpoint),
```

Comment:
Nit: You could change the `userAssert` above into an `user().assertString` to avoid the double assert.